### PR TITLE
[2/n][dagster-airbyte] Scaffold DagsterAirbyteTranslator for rework

### PR DESCRIPTION
**Pull request by maximearmstrong**
## Summary & Motivation

Builds out a very barebones translator and props classes for the new version of the Airbyte cloud integration.

## How I Tested These Changes

Tests will be added in subsequent PRs.

### CODE REVIEW

**Comment by maximearmstrong** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26205?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#26205** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26205?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26205?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#26204** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26204?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by dpeng817** [comment]
I'm assuming the upstack is meant to be non-draft as well?

**Review by dpeng817 — CHANGES_REQUESTED** [review]
I think this needs to wait on approval on downstreams since this PR doesn't have full context. Removing from my queue for now.

**Review by dpeng817 — APPROVED** [review]

**Pull request merged**
